ngraph/core/reference/include/ngraph/runtime/reference/group_convolution_backprop_data.hpp

```diff
@@ -25,6 +25,127 @@ namespace ngraph
     {
         namespace reference
         {
```

Comment:
Added validation functions to check input ranks, spatial dimensions of strides, dilations and pads. Additionally output shape is inferred and compared to output shape provided by the tensor descriptor. These checks are introduced because when using dynamic shapes some node validation check from `validate_and_infer_types` are by-passed.

`infer_backward_conv_output_shape` and `validate_convolution_backprop_data_parameters` to be moved to ConvolutionbackpropData reference implementation once it is refactored.